src/r2mm/manager/PackageDexieStore.ts

```diff
@@ -0,0 +1,72 @@
+import Dexie, { Table } from 'dexie';
+
+import ThunderstoreMod from '../../model/ThunderstoreMod';
+
+interface DexieVersion {
+    full_name: string;
+    name: string;
+    version_number: string;
+    uuid4: string;
+    dependencies: string[];
+    description: string;
+    icon: string;
+    is_active: boolean;
+    downloads: number;
+    download_url: string;
+    website_url: string;
+    file_size: number;
+    date_created: Date;
+}
+
+interface DexiePackage {
+    full_name: string; // Primary key
+    owner: string;
+    name: string;
+    community: string; // Not included in the API response
+    uuid4: string;
+    package_url: string;
+    categories: string[];
+    rating_score: number;
+    is_pinned: boolean;
+    is_deprecated: boolean;
+    has_nsfw_content: boolean;
+    donation_link: string | undefined;
+    date_created: Date;
+    date_updated: Date;
+    versions: DexieVersion[];
+}
+
+class PackageDexieStore extends Dexie {
+    packages!: Table<DexiePackage, string>;
+
+    constructor() {
+        super('tsPackages');
+
+        this.version(1).stores({
+            packages: '[community+full_name]'
+        });
+    }
+}
+
+const db = new PackageDexieStore();
+
+// TODO: user type guards to validate (part of) the data before operations?
+export async function updateFromApiResponse(community: string, packages: any[]) {
+    const newPackages: DexiePackage[] = packages.map((pkg) => ({...pkg, community}));
+
+    await db.transaction(
+        'rw',
+        db.packages,
+        async () => {
+            // .bulkDelete is faster than calling .delete() on the collection directly.
+            const oldIds = await db.packages.where({community}).primaryKeys();
+            await db.packages.bulkDelete(oldIds);
+            await db.packages.bulkAdd(newPackages)
+        }
+    );
+};
```

Comment:
I didn't future proof or speculate on this since I didn't know how the API end will end up being changed. What you suggest SGTM though, since getting rid of it won't be a big deal if we end up going some other way. Who knows, maybe "UPSERT + DELETE outdated" can be more performant than "DELETE all + INSERT" even when the mod list is received in one go?

Reply:
I also realized dexie doesn't have a table schema so adding it later on would also work, review comment was based on the assumption that it would need a migration of some kind if done later. But as you noted, the perf gain might be significant enough to warrant doing it already.

Reply:
Wasn't that straightforward to judge which way is more performant since it seems to depend on the size of the package list. [Changed to "UPSERT + DELETE"](https://github.com/ebkr/r2modmanPlus/pull/1261/commits/ddcd60b767ff8142f30b540b5f7745f0e20cc235#diff-2122463b05e5c9342979622ca58d48970d706008d1d9df423063e69e07055988R36-R87) anyway since it had some slight benefits.